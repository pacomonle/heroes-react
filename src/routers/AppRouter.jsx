import React, { useContext } from 'react'
import {
    BrowserRouter as Router,
    Switch,
  } from "react-router-dom";
import { AuthContext } from '../auth/AuthContext.js';

import LoginScreen from '../components/Login/LoginScreen.jsx';
import DashBoardRoutes from './DashBoardRoutes.jsx';
import PrivateRoutes from './PrivateRoutes.jsx';
import PublicRoutes from './PublicRoutes.jsx';



const AppRouter = () => {

const {user:{logged}} = useContext(AuthContext)

    return (
        <Router>
            <div>
                <Switch>
                    <PublicRoutes isAutenticated={logged} exact path='/login' component={LoginScreen} />
     {/* para las rutas hijas va sin el exact para recoger todo lo que va detras del '/' */}
                    <PrivateRoutes isAutenticated={logged} path='/' component={DashBoardRoutes} />
                </Switch>
            </div>
        </Router>


    )
}

export default AppRouter
