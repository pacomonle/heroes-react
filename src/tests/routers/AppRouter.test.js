import { mount } from 'enzyme';
import React from 'react';
import { AuthContext } from '../../auth/AuthContext';
import AppRouter from '../../routers/AppRouter';

describe('Pruebas en el AppRouter', () => {

   // const {user:{logged}} = useContext(AuthContext)
   const contextValue = {
       dispatch: jest.fn(),
       user:{
        logged: false
       }     
   }

    test('debe de mostrar login si no está autenticado ', () => {
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );
        expect( wrapper ).toMatchSnapshot();
        
    })

    test('debe de mostrar el componente marvel si está autenticado', () => {
        
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Juan'
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper.find('.navbar').exists() ).toBe(true);
        

    })
})
