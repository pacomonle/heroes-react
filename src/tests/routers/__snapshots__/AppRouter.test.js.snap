// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el AppRouter debe de mostrar login si no est√° autenticado  1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
  </div>
</div>
`;
